src/run.rs

```diff
@@ -266,22 +267,49 @@ pub fn clean_up(
     tmp_dir: TempDir,
     worktree_branch: Option<&str>,
     subprocesses: &mut [Child],
+    prompt: bool,
 ) -> Result<()> {
-    if !delete_file.exists() {
+    let mut delete = delete_file.exists();
+
+    if prompt && delete {
+        println!("Are you sure you want to delete this project? (Y/n)");
+        loop {
+            let mut input = String::new();
+
+            match stdin().read_line(&mut input) {
+                Ok(_n) => match input.trim() {
+                    "" | "Yes" | "yes" | "Y" | "y" => break,
+                    "No" | "no" | "N" | "n" => {
+                        let _ = fs::remove_file(&delete_file);
```

Comment:
This should be done later in the code that does the preservation.

Try to keep the kind of logic together:
- something that does ask for input from the user should only do that
- something that does the actual job of deleting or preserving should do just that

Otherwise you're creating "side effects" and it makes your code more dangerous for the next developer who will need to touch it.

Example: let's say it turns out we want to delete the prompt because we don't like it anymore. The developer will go into the code, take the whole paragraph, delete it, done... this will work only if your code was doing exactly just that.

Reply:
Yeah but I need to delete the file myself only in that case no?

```
we check if the TO_DELETE file exists

If it's still there and I use the prompt, I print the prompt and `delete` is set on `true`
    I wait for an input
    * if it's yes, I break with `delete` on true.
    * if it's no, I remove the `TO_DELETE` file and I break with `delete` on false.
    * if `_ =>` I wait for another input from the user and the loop loops. 

// normal behavior
if the `TO_DELETE` file exists (or `delete == true`), we don't keep the project
If it doesn't exists, we keep the project
```

So if we want to get rid of the prompt this wouldn't be a problem since the behavior without prompt doesn't need to delete the `TO_DELETE` file.

Reply:
So this code:
```
let mut delete = delete_file.exists();

if prompt && delete {
    // ...
}

if delete {
    // ...
} else {
    // ...
}
```
Can be replaced by:
```
if delete_file.exists() {
    // ...
} else {
    // ...
}
```

Reply:
> Yeah but I need to delete the file myself only in that case no?

When preserving, always delete the TO_DELETE file, doesn't matter if it exists or not. So it will cover both scenarios.